src/cast_core/src/CrudActor.cpp

```diff
@@ -305,14 +379,16 @@ struct UpdateOneOperation : public WriteOperation {
     void run(mongocxx::client_session& session) override {
         auto filter = _filterExpr();
         auto update = _updateExpr();
-        auto ctx = _operation.start();
-        auto result = (_onSession)
-            ? _collection.update_one(session, std::move(filter), std::move(update), _options)
-            : _collection.update_one(std::move(filter), std::move(update), _options);
-        if (result) {
-            ctx.addDocuments(result->modified_count());
-        }
-        ctx.success();
+        this->doBlock(_operation, [&](metrics::OperationContext& ctx) {
+            auto result = (_onSession)
+                ? _collection.update_one(session, filter.view(), update.view(), _options)
```

Comment:
Henrik pointed out you could pass in a reference to `nullptr` for `session` to have it not use a session. It's how the `CommitLatency` workload works.

Reply:
Probably not worth doing it here though

Reply:
Cool....that would save an `if` statement in pretty much every operation but it feels like dangerously undocumented behavior..so I'll leave it alone for now 😸